session/filesync/filesync.go

```diff
@@ -230,47 +232,102 @@ func FSSync(ctx context.Context, c session.Caller, opt FSSendRequestOpt) error {
 	return pr.recvFn(stream, opt.DestDir, opt.CacheUpdater, opt.ProgressCb, opt.Differ, opt.Filter)
 }
 
-// NewFSSyncTargetDir allows writing into a directory
-func NewFSSyncTargetDir(outdir string) session.Attachable {
-	p := &fsSyncTarget{
+type FSSyncTarget interface {
+	target() *fsSyncTarget
+}
+
+type fsSyncTarget struct {
+	id     string
+	outdir string
+	f      FileOutputFunc
+}
+
+func (target *fsSyncTarget) target() *fsSyncTarget {
+	return target
+}
+
+func WithFSSync(id string, f FileOutputFunc) FSSyncTarget {
```

Comment:
Not a blocker but a variadic opt `WithID` would have been nice.

Reply:
Alternatively. What do you think about the idea of removing the ID entirely? And just using the index in the output entries array?

Reply:
This could work if the output file/output directory could be used as IDs. For this, the `FileOutputFunc` could become an interface and would identify itself: for files this would the file path (currently not persisted once the output has been "resolved"), with a directory - likewise. This would also allow for better automatic error messages without the need from the user to provide custom IDs.

Reply:
Have updated to remove the ID and just use the index of the exporter - I think this is actually a lot neater and requires less specific configuration.